### PR TITLE
[CBRD-24477] add pl/csql extended cases

### PR DESCRIPTION
**Pull request by hiclass**
http://jira.cubrid.org/browse/CBRD-24477
add pl/csql extended cases
on 11.4.0.1328-d1fd8de

### CODE REVIEW

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_ascii.sql [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 create or replace procedure t () as
 begin
-    -- dbms_output.put_line(ASCII(NULL)); -- param NULL parse error
+    dbms_output.put_line(ASCII(NULL)); -- param NULL parse error
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_ascii.sql:7]: Problem of single NULL argument to a built-in function is resolved in issue [CBRD-25168]. 
This kind of comments can be erased in this and other lines in TCs

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_04_type_convert_function/cases/04_04_03_04_bfn_type_cast1.sql [hunk 2]
```diff
@@ -5,7 +5,6 @@
 create or replace procedure t () as
 begin
     -- CAST( AS ) parse error
-    /*
     dbms_output.put_line('INTEGER');
     dbms_output.put_line(CAST(NULL AS INTEGER));
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_04_type_convert_function/cases/04_04_03_04_bfn_type_cast1.sql:9]: Currently, unique syntax of CAST is not supported.
This TC and the other one that test CAST should be removed.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_05_datetime_function/cases/04_04_03_05_bfn_datetime_extract.sql [hunk 3]
```diff
@@ -5,7 +5,6 @@
 create or replace procedure t () as
 begin
     -- parse error about EXTRACT()
-    /*
     dbms_output.put_line(EXTRACT(NULL));
     dbms_output.put_line(EXTRACT(MILLISECOND FROM DATETIME '1999-01-11 12:34:52.333'));
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_05_datetime_function/cases/04_04_03_05_bfn_datetime_extract.sql:9]: Currently, unique syntax of EXTRACT is not supported.
This TC should be removed.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_06_information_function/cases/04_04_03_06_bfn_info_typeof.sql [hunk 4]
```diff
@@ -5,11 +5,8 @@
 create or replace procedure t () as
 begin
     -- parse error for TYPEOF() 
-    /*
     dbms_output.put_line(TYPEOF(1));
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_06_information_function/cases/04_04_03_06_bfn_info_typeof.sql:8]: Currently, typeof function is not supported.
This TC should be removed for now, and added later.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_bit_length.sql [hunk 5]
```diff
@@ -4,11 +4,11 @@
...
 create or replace procedure t () as
 begin
-    -- dbms_output.put_line(BIT_LENGTH(NULL)); -- parse error NULL
-    -- dbms_output.put_line(BIT_LENGTH('')); -- param error '', Semantic: Cannot coerce host var to type bit varying.
-    -- dbms_output.put_line(BIT_LENGTH('CUBRID')); -- param type mismath 
+    dbms_output.put_line(BIT_LENGTH(NULL)); -- parse error NULL
+    dbms_output.put_line(BIT_LENGTH('')); -- param error '', Semantic: Cannot coerce host var to type bit varying.
+    dbms_output.put_line(BIT_LENGTH('CUBRID')); -- param type mismath 
     dbms_output.put_line(BIT_LENGTH('큐브리드'));
-    -- dbms_output.put_line(BIT_LENGTH(B'01010101')); -- parse error type literal define parenthesis
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_bit_length.sql]: Literals of the form B'...' are not supported. 
This and other lines that use those literals should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_chr.sql [hunk 6]
```diff
@@ -4,12 +4,12 @@
 
 create or replace procedure t () as
 begin
-    -- dbms_output.put_line(CHR(NULL)); -- param NULL parse error
+    dbms_output.put_line(CHR(NULL)); -- param NULL parse error
     dbms_output.put_line(CHR(68));
     dbms_output.put_line(CHR(68-2));
-    -- dbms_output.put_line(CHR(14909886 USING utf8)); -- parse error for USING keyord and RPAREN
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_chr.sql]: Currently, unique syntax of CHR is not supported.
This line and others that use the syntax should be erased

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_chr.sql [hunk 7]
```diff
@@ -4,12 +4,12 @@
...
-    -- dbms_output.put_line(CHR(_utf8'ま')); -- parse error type literal define parenthesis
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_chr.sql]: Currently, literals of the form _utf8'...' are not supported.
This line and others that use those literals should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_concat.sql [hunk 8]
```diff
@@ -4,13 +4,13 @@
 
 create or replace procedure t () as
 begin
-    -- dbms_output.put_line(CONCAT(NULL)); -- param NULL parse error for first param
+    dbms_output.put_line(CONCAT(NULL)); -- param NULL parse error for first param
     dbms_output.put_line(CONCAT('CUBRID ', '2024 ' , '11.4'));
     dbms_output.put_line(CONCAT('CUBRID ', '2024 ' , 11.4));
-    -- dbms_output.put_line(CONCAT('CUBRID ', NULL , 11.4)); -- param NULL parse error
-    -- dbms_output.put_line(CONCAT(14909886 USING utf8)); -- parse error for USING keyord and RPAREN
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_concat.sql]: Currntly, USING in CONCAT is not supported.
This line should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_position.sql [hunk 9]
```diff
@@ -4,14 +4,12 @@
 
 create or replace procedure t () as
 begin
-    -- dbms_output.put_line(POSITION(NULL)); -- param NULL parse error for first param
+    dbms_output.put_line(POSITION(NULL)); -- param NULL parse error for first param
     -- parse error POSITON() of IN keword
-    /*
     dbms_output.put_line(POSITION('5' IN '12345abcdeabcde'));
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_position.sql:9]: Currently, IN in POSITION is not supported.
This and other lines that uses it should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_replace.sql [hunk 10]
```diff
@@ -12,7 +12,7 @@ begin
     dbms_output.put_line(REPLACE('CUBRID DATABASE', 3.0));
     dbms_output.put_line(REPLACE('CUBRID DATABASE', 32000000));
     dbms_output.put_line(REPLACE('你好我好大家好','好','开心'));
-    -- dbms_output.put_line(REPLACE(_binary'abcdefg', 'abcd')); -- parse error type literal define parenthesis
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_replace.sql]: Currently, literals of the form _binary'...' are not supported.
This and other lines that uses those literals should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_strcmp.sql [hunk 11]
```diff
@@ -10,7 +10,7 @@ begin
     dbms_output.put_line(STRCMP('abc', NULL));
     dbms_output.put_line(STRCMP('abcd', 'abc'));
     dbms_output.put_line(STRCMP('abc', 'abcd'));
-    --dbms_output.put_line(STRCMP('abc' COLLATE utf8, 'abc' COLLATE utf8_en_ci )); -- parse error COLLATE keyword
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_strcmp.sql]: Currently, COLLATE of string literals is not supported.
This and other lines that uses it should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_substring.sql [hunk 12]
```diff
@@ -11,12 +11,10 @@ begin
     dbms_output.put_line(SUBSTRING('12345abcdeabcde',6, -4));
 
     -- parse error SUBSTRING() in FROM/FOR
-    /*
     dbms_output.put_line(SUBSTRING('12345abcdeabcde' FROM 6 FOR 4));
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_substring.sql:14]: Currently, unique syntax of SUBSTRING is not supported.
This and other lines that uses it should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_trim.sql [hunk 13]
```diff
@@ -4,16 +4,14 @@
 
 create or replace procedure t () as
 begin
-    -- dbms_output.put_line(TRIM(NULL)); -- param NULL parse error for first param
+    dbms_output.put_line(TRIM(NULL)); -- param NULL parse error for first param
     dbms_output.put_line('[' || TRIM('   CUBRID   ') || ']');
 
     -- TRIM() keyword FROM/LEADING/TRAILING/BOTH parse error
-    /*
     dbms_output.put_line(TRIM('a' FROM 'aaaCUBRIDaaaa'));
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_trim.sql:11]: Currently, unique syntax of TRIM is not supported.
This and other lines that uses it should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_02_numeric_function/cases/04_04_03_02_bfn_numeric_truncate.sql [hunk 14]
```diff
@@ -4,7 +4,7 @@
 
 create or replace procedure t () as
 begin
-    --dbms_output.put_line(TRUNCATE(0.4)); -- TRUNCATE word parse error. might be confused with TRUNCATE TABLE
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_02_numeric_function/cases/04_04_03_02_bfn_numeric_truncate.sql]: TRUNCATE takes two arguments.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_05_datetime_function/cases/04_04_03_05_bfn_datetime_adddate.sql [hunk 15]
```diff
@@ -8,7 +8,6 @@ begin
     dbms_output.put_line(ADDDATE(TO_DATETIME('1999-01-08'), NULL));
 
     -- ADDDATE() INTERVAL delimiter '' and option parse error
-    /*
     dbms_output.put_line(ADDDATE(NULL, INTERVAL 123 MILLISECOND));
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_05_datetime_function/cases/04_04_03_05_bfn_datetime_adddate.sql:11]: Currently, INTERVAL in ADDDATE, DATE_SUB, SUBDATE, etc  is not supported.
This and other lines that uses it should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_05_datetime_function/cases/04_04_03_05_bfn_datetime_months_between.sql [hunk 16]
```diff
@@ -4,8 +4,8 @@
 
 create or replace procedure t () as
 begin
-    -- dbms_output.put_line(MONTHS_BETWEEN(NULL, NULL)); -- parse NULL param error
-    -- dbms_output.put_line(MONTHS_BETWEEN(TO_DATETIME('1999-01-11 12:34:52.333'), NULL)));  -- parse NULL param error
+    dbms_output.put_line(MONTHS_BETWEEN(NULL, NULL)); -- parse NULL param error
+    dbms_output.put_line(MONTHS_BETWEEN(TO_DATETIME('1999-01-11 12:34:52.333'), NULL)));  -- parse NULL param error
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_05_datetime_function/cases/04_04_03_05_bfn_datetime_months_between.sql:8]: The last ')' is unmatched.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_05_datetime_function/cases/04_04_03_05_bfn_datetime_timediff.sql [hunk 17]
```diff
@@ -14,8 +13,6 @@ begin
     dbms_output.put_line(TIMEDIFF(TO_TIME('12:34:52'), TO_TIME('00:00:00')));
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_05_datetime_function/cases/04_04_03_05_bfn_datetime_timediff.sql:13]: Type error in the above line. 
TIMEDIFF(TO_DATE('1999-01-11'), TO_TIME('23:59:59'))

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_07_encryption_function/cases/04_04_03_07_bfn_encrypt_sha2.sql [hunk 18]
```diff
@@ -4,15 +4,11 @@
...
-    -- parse error for SHA2()
-    /*
-    dbms_output.put_line(SHA2('cubrid'));
-    dbms_output.put_line(SHA2(255));
-    dbms_output.put_line(SHA2('01/01/2010'));
-    dbms_output.put_line(SHA2(TO_DATE('01/01/2010')));
-    */
-    null;
+    dbms_output.put_line(SHA2(NULL)); -- param NULL parse error for first param
+    dbms_output.put_line(SHA2('cubrid', 224); -- parse error for SHA2() param desired length
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_07_encryption_function/cases/04_04_03_07_bfn_encrypt_sha2.sql:8]: The first '(' is not matched in this and the next lines.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_07_encryption_function/cases/04_04_03_07_bfn_encrypt_sha2.sql [hunk 19]
```diff
@@ -0,0 +1,21 @@
+--+ server-message on
+
+-- normal: basic usage of a builtin function call
+
+create or replace procedure t () as
+begin
+    dbms_output.put_line(SHA2(NULL)); -- param NULL parse error for first param
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_07_encryption_function/cases/04_04_03_07_bfn_encrypt_sha2.sql:7]: SHA2 must have two arguments

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_08_comparision_function/cases/04_04_03_08_bfn_comp_if.sql [hunk 20]
```diff
@@ -0,0 +1,50 @@
+--+ server-message on
+
+-- normal: basic usage of a builtin function call
+
+create or replace procedure t () as
+begin
+    -- function not work and skip IF()
+    dbms_output.put_line('-- IF simple compare');	
+    dbms_output.put_line(IF(NULL,  NULL, NULL));
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_08_comparision_function/cases/04_04_03_08_bfn_comp_if.sql:9]: IF built-in function cannot be used due to a parsing problem.
This TC should be removed.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_position.sql [hunk 21]
```diff
@@ -0,0 +1,22 @@
+--+ server-message on
+
+-- normal: basic usage of a builtin function call
+
+create or replace procedure t () as
+begin
+    dbms_output.put_line(POSITION(NULL)); -- param NULL parse error for first param
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_01_string_function/cases/04_04_03_01_bfn_string_position.sql:7]: Syntax error. 
This line should be erased.

File: sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_06_information_function/cases/04_04_03_06_bfn_info_inet.sql [hunk 22]
```diff
@@ -0,0 +1,21 @@
+--+ server-message on
+
+-- normal: basic usage of a builtin function call
+
+create or replace procedure t () as
+begin
+    dbms_output.put_line(INET_ATON(NULL)); -- param NULL parse error for first paramA
+    dbms_output.put_line(INET_NTOA(NULL)); -- param NULL parse error for first paramA
```
Comment (hyunikn) [inline sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_06_information_function/cases/04_04_03_06_bfn_info_inet.sql:8]: INET_NTOA(NULL) causes a runtime error in SQL statement "select INET_NTOA(NULL);" too.
This needs to be resolved in a separate Jira issue.

**Comment by hgryoo** [comment]
For regression testing on Circle CI, I'm going to merge now. I'd like to reflect the reviews on the PR merging into develop branch.

**Pull request merged**